rules/windows/process_creation/proc_creation_win_cmd_for_tokens_dir.yml

```
@@ -0,0 +1,32 @@
+title: Suspicious Use of for Loop with Directory Search in CMD
+id: 2782fbd8-b662-4eb5-9962-5bfbfb671e7b
+status: experimental
+description: Detects suspicious usage of the cmd.exe for /f loop combined with the tokens= parameter and a recursive directory listing. This pattern may indicate an attempt to discover and execute system binaries dynamically for example powershell, a technique sometimes used by attackers to evade detections.
+references:
+    - https://www.virustotal.com/gui/file/29837d0d3202758063185828c8f8d9e0b7b42b365c8941cc926d2d7c7bae2fb3
+    - https://www.virustotal.com/gui/search/behavior%253A%2522for%2520%252Ff%2520tokens%253D*%255C%2522%2520%2525a%2520in%2520('dir%2522?type=files # behavior:"for /f tokens=*\" %a in ('dir"
+author: Joseliyo Sanchez, @Joseliyo_Jstnk
+date: 2025-07-10
+tags:
+    - attack.execution
+    - attack.t1059.003
+logsource:
+    category: process_creation
+    product: windows
+detection:
+    selection_tokens:
+        CommandLine|contains|all:
+            - 'for /f'
+            - 'tokens='
+            - 'in ('
+            - 'dir'
+    selection_tokens_parent:
+        ParentCommandLine|contains|all:
+            - 'for /f'
+            - 'tokens='
+            - 'in ('
+            - 'dir'
+    condition: 1 of selection_*
```

```suggestion
detection:
    selection:
        ParentImage|endswith: '\explorer.exe'
        CommandLine|contains|all:
            - 'for /f'
            - 'tokens='
            - 'in ('
            - 'dir'
    condition: selection
```
This kind of behavior is commonly observed during the execution of malicious .lnk files, so I believe keeping the parent image as Explorer.exe is appropriate.  Also I did some research and found various legit programs using similar pattern, so relying solely on the CommandLine for detection may result in false positives.

---

I'm not sure this is the best idea. If you look at the Sysmon event I left above, the parent is cmd.exe, not explorer.exe.

I've been checking some of the LNKs from the VT query I left above and didn't see any children of explorer.exe. The thing is, that command can primarily be executed by cmd.exe or powershell.exe.

IMHO, I'd leave it as is :)

---

In VT, .lnk files are executed dynamically by PowerShell or CMD, as there is no direct user interaction involved. However, in a real-world scenario, a user must double-click the .lnk file to open it. We could consider adding CMD or PowerShell as a parent image, but these strings used in the logic are not highly specific and may result in false positives.